cosmwasm/enclaves/execute/src/registration/onchain.rs

```diff
@@ -7,19 +7,83 @@ use std::panic;
 use enclave_ffi_types::NodeAuthResult;
 
 use crate::registration::seed_exchange::SeedType;
+use crate::registration::attestation::verify_quote_ecdsa;
+use crate::registration::cert::verify_ra_report;
+
 use enclave_crypto::consts::OUTPUT_ENCRYPTED_SEED_SIZE;
 use enclave_crypto::PUBLIC_KEY_SIZE;
 use enclave_utils::{
     oom_handler::{self, get_then_clear_oom_happened},
     validate_const_ptr, validate_mut_ptr,
 };
 
+use sgx_types::sgx_ql_qv_result_t;
+
+#[cfg(feature = "SGX_MODE_HW")]
+use enclave_crypto::consts::SigningMethod;
+
 use super::cert::verify_ra_cert;
 use super::seed_exchange::encrypt_seed;
+use core::mem;
+use std::slice;
 
 #[cfg(feature = "light-client-validation")]
 use enclave_contract_engine::check_cert_in_current_block;
 
+#[cfg(feature = "light-client-validation")]
+use block_verifier::VERIFIED_BLOCK_MESSAGES;
+
+#[cfg(feature = "light-client-validation")]
+fn get_current_block_time_s() -> i64
+{
+    let verified_msgs = VERIFIED_BLOCK_MESSAGES.lock().unwrap();
+    let tm_ns = verified_msgs.time();
+    return (tm_ns / 1000000000) as i64;
+}
+
+#[cfg(not(feature = "light-client-validation"))]
+fn get_current_block_time_s() -> i64
+{
+    return 0 as i64;
+}
+
+
+pub fn split_combined_cert(
```

Comment:
Can this function be private? Also why not just pass the cert_slice in here? You should avoid using raw pointers wherever possible